File: Basic Programming/src/Coin Game.java
import java.io.*;
import java.util.*;

public class Main {

    static class FastScanner {
        private BufferedReader br;
        private StringTokenizer st;

        FastScanner() {
            br = new BufferedReader(new InputStreamReader(System.in));
        }

        String next() throws IOException {
            while (st == null || !st.hasMoreTokens())
                st = new StringTokenizer(br.readLine());
            return st.nextToken();
        }

        int nextInt() throws IOException {
            return Integer.parseInt(next());
        }
    }

    public static void main(String[] args) throws Exception {

        FastScanner fs = new FastScanner();
        PrintWriter out = new PrintWriter(System.out);

        int t = fs.nextInt();

        while (t-- > 0) {

            int n = fs.nextInt();
            int[] A = new int[n];

            for (int i = 0; i < n; i++) {
                A[i] = fs.nextInt();
            }

            int c = 0;

            for (int i = 0; i < n; i++) {
                while (A[i] % 2 == 0) {
                    c++;
                    A[i] /= 2;
                }
            }

            if (c % 2 == 0)
                out.println("Alan");
            else
                out.println("Charlie");
        }

        out.flush();
    }
}

File: Basic Programming/src/Coin Game.swift
import Foundation

let t = Int(readLine()!)!

for _ in 0..<t {

    let n = Int(readLine()!)!
    var A = readLine()!.split(separator: " ").map { Int($0)! }

    var c = 0

    for i in 0..<n {
        var x = A[i]
        while x % 2 == 0 {
            c += 1
            x /= 2
        }
    }

    if c % 2 == 0 {
        print("Alan")
    } else {
        print("Charlie")
    }
}
